csharp/extractor/Semmle.Extraction/Options.cs

```diff
@@ -64,6 +66,16 @@ public virtual bool HandleOption(string key, string value)
                 case "verbosity":
                     Verbosity = (Verbosity)int.Parse(value);
                     return true;
+                case "compression":
+                    try
+                    {
+                        TrapCompression = (TrapWriter.CompressionMode)Enum.Parse(typeof(TrapWriter.CompressionMode), value, true);
+                        return true;
+                    }
+                    catch (ArgumentException)
+                    {
+                        return false;
+                    }
```

Comment:
You could replace this with `TryParse<TEnum>(String, Boolean, TEnum)`: https://docs.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=net-6.0#system-enum-tryparse-1(system-string-system-boolean-0@), which would look a bit nicer.